pipeline/transformer/transformer_calm/src/main/scala/uk/ac/wellcome/platform/transformer/calm/TransformerWorker.scala

```diff
@@ -0,0 +1,95 @@
+package uk.ac.wellcome.platform.transformer.calm
+
+import akka.{Done, NotUsed}
+import akka.stream.scaladsl.{Flow, Sink, Source}
+import com.amazonaws.services.sqs.model.Message
+import uk.ac.wellcome.bigmessaging.BigMessageSender
+import uk.ac.wellcome.json.JsonUtil._
+import uk.ac.wellcome.messaging.sns.NotificationMessage
+import uk.ac.wellcome.messaging.sqs.SQSStream
+import uk.ac.wellcome.models.work.internal.TransformedBaseWork
+import uk.ac.wellcome.storage.store.VersionedStore
+import uk.ac.wellcome.storage.{Identified, Version}
+
+import scala.concurrent.Future
+
+sealed abstract class CalmWorkerError(msg: String) extends Exception(msg)
+case class DecodeKeyError(msg: String) extends CalmWorkerError(msg)
+case class StoreReadError(msg: String) extends CalmWorkerError(msg)
+case class TransformerError(msg: String) extends CalmWorkerError(msg)
+case class MessageSendError(msg: String) extends CalmWorkerError(msg)
+
+/**
+  * A transformer worker:
+  * - Takes an SQS stream that emits VHS keys
+  * - Gets the record of type `In`
+  * - Runs it through a transformer and transforms the `In` to `TransformedBaseWork`
+  * - Emits the message via `BigMessageSender` to an SNS topic
+  */
+trait TransformerWorker[In, SenderDest] {
+  type StreamMessage = (Message, NotificationMessage)
+  type Result[T] = Either[Throwable, T]
+  type StoreKey = Version[String, Int]
+
+  val name: String
+  val stream: SQSStream[NotificationMessage]
+  val sender: BigMessageSender[SenderDest, TransformedBaseWork]
+  val store: VersionedStore[String, Int, In]
+  val transformer: Transformer[In]
+
+  val errorSink: Sink[Result[_], Future[Done]] = Sink.ignore
```

Comment:
What do we want to happen to these errors? Maybe at lease log them?

```suggestion
  def errorSink: Sink[Result[_], Future[Done]] =
    Flow[Result[_], NotUsed].collect { case Left(err) => logError(err) }.to(Sink.ignore)
  
  def logError(err: Throwable) =
    error(s"Exception in worker $name", err)
```